src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzer.cs

```diff
@@ -490,10 +490,20 @@ private static void ReportDiagnostics(IOperation operation, SmallDictionary<stri
                 IInvocationOperation iOperation => iOperation.TargetMethod,
                 IObjectCreationOperation cOperation => cOperation.Constructor,
                 IFieldReferenceOperation fOperation => IsWithinConditionalOperation(fOperation) ? null : fOperation.Field,
+                IPropertyReferenceOperation pOperation => GetPropertyAccessor(pOperation.Property, operation),
                 IMemberReferenceOperation mOperation => mOperation.Member,
                 _ => null,
             };
 
+        private static ISymbol GetPropertyAccessor(IPropertySymbol property, IOperation operation)
+            => operation.GetValueUsageInfo(property.ContainingSymbol) switch
+            {
+                ValueUsageInfos.Read => property.GetMethod,
```

Comment:
Using a switch statement to compare against a single field may not work correctly when the return flags value has multiple bits set, for example `ValueUsageInfo.ReadableReference` (`in` param), `ValueUsageInfo.WritableReference` (`out` param) and `ValueUsageInfo.ReadableWritableReference`  (`ref` param). Instead it would be better to write this method as:
```csharp
private static IEnumerable<IMethodSymbol> GetReferencedAccessorSymbols(IPropertySymbol property, IPropertyReferenceOperation operation)
{
    var valueUsageInfo = operation.GetValueUsageInfo(property.ContainingSymbol);
    if (valueUsageInfo & ValueUsageInfo.Read != 0 || valueUsageInfo & ValueUsageInfo.Name != 0)
    {
        yield return property.GetMethod;
    }

    if (valueUsageInfo & ValueUsageInfo.Write != 0)
    {
        yield return property.SetMethod;
    }
}
`

```

Reply:
You may also want to copy over https://github.com/dotnet/roslyn/blob/755c3b186fba57b676b5db41d7dce09b600f1e42/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/ValueUsageInfo.cs#L67-L80 to make the analyzer code cleaner :-)